clippy_lints/src/macros_hiding_unsafe_code.rs

```diff
@@ -0,0 +1,140 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, in_macro, is_lint_allowed, source::snippet_with_applicability};
+use rustc_errors::Applicability;
+use rustc_hir::{intravisit::FnKind, Block, BlockCheckMode, Body, FnDecl, FnSig, HirId, UnsafeSource, Unsafety};
+use rustc_lint::{
+    builtin::{UNSAFE_OP_IN_UNSAFE_FN, UNUSED_UNSAFE},
+    LateContext, LateLintPass, LintContext,
+};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for macro calls inside an `unsafe` function which expansion
+    /// contains an `unsafe` block, while the macro call is not wrapped in an `unsafe` block
+    /// itself. This lint only triggers in functions where the `unsafe_op_in_unsafe_fn` lint is
+    /// enabled.
+    ///
+    /// **Why is this bad?** This hides an unsafe operation inside a macro call. This is against
```

Comment:
I don't think that being against the intention of lint is a strong motivation since it's the nature of the lint to catch what it's supposed to catch. Like, chicken and egg problem.

I'd emphasize that the macro containing an unsafe can hide the fact that it uses unsafe, and thus makes it easier to use in the context where required invariants are not held.

Reply:
Good point, I'll word this differently.